app/views/catalog/_index_header_gallery_collection.html.erb

```diff
@@ -1,4 +1,5 @@
 <div class="search-results-title-row">
   <h4 class="search-result-title"><%= link_to document.title_or_label.truncate_words(5), [hyrax, document] %></h4>
+  <%= label_tag "Add to My Shelf", raw("#{check_box_tag "my_shelf"} Add to My Shelf") %>
```

Comment:
The fact that these passed axe-matcher is worrying. The labels have no matching elements (you should be able to click the label and it will check the box), and checkboxes with repeated IDs.

This might work better
```
  <%= label_tag "" do %>
    <% raw("#{check_box_tag document.id} Add to My Shelf") %>
  <% end %>
```